### PR TITLE
Fix mo serialization (QC-183)

### PR DESCRIPTION
**Pull request by Barthelemy**

### CODE REVIEW

**Comment by Barthelemy** [comment]
Remember to remove the two extra commits.

File: Framework/test/testMonitorObject.cxx [hunk 1]
```diff
@@ -61,4 +67,48 @@ BOOST_AUTO_TEST_CASE(mo_check)
...
+  cout << "quality : " << obj.getQuality() << endl;
+  cout << "check numbers : " << obj.getChecks().size() << endl;
+  CheckDefinition c = obj.getCheck("name2");
+  cout << "check2 libraryName : " << c.libraryName << endl;
+  TFile file("/tmp/test.root", "RECREATE");
+  obj.Write(obj.getName().data());
+  file.Close();
+
+  cout << "***" << endl;
+  TFile file2("/tmp/test.root");
```
Comment (knopers8) [inline Framework/test/testMonitorObject.cxx]: If there are several builds/test running on the same CI machine, wouldn't that cause some race conditions? Though I don't have an idea how to solve it and anyway they would be very rare probably. Maybe a random number in the name?

Comment (Barthelemy) [reply Framework/test/testMonitorObject.cxx]: A random number would do it. I will create a separate PR so that this one can go ahead

Comment (Barthelemy) [reply Framework/test/testMonitorObject.cxx]: Thank you , this is a good remark.

**Pull request merged**